libs/agno/agno/db/mongo/async_mongo.py

```diff
@@ -86,17 +191,30 @@ def __init__(
             culture_table=culture_collection,
         )
 
+        # Detect client type if provided
+        if db_client is not None:
+            self._client_type = _detect_client_type(db_client)
+            if self._client_type == self.CLIENT_TYPE_UNKNOWN:
+                # Default to PyMongo async if available, else Motor
+                self._client_type = (
+                    self.CLIENT_TYPE_PYMONGO_ASYNC if PYMONGO_ASYNC_AVAILABLE else self.CLIENT_TYPE_MOTOR
+                )
+        else:
+            # Auto-select preferred library when creating from URL
+            # Prefer PyMongo async if available, fallback to Motor
+            self._client_type = self.CLIENT_TYPE_PYMONGO_ASYNC if PYMONGO_ASYNC_AVAILABLE else self.CLIENT_TYPE_MOTOR
+
```

Comment:
- It'd be cleaner to resolve imports here. If db_client is present, we check its type, and we import pymongo/motor based on that. If none we raise saying we don't support the client.
- We don't want to prio PyMongo, but use whatever the user is passing
- When you need to use the un-imported types, you can just do `"AsyncIOMotorClient"` etc, so you don't need the AsyncMongoClientType class before initialization (because at that point we don't know what we are gonna import)